### PR TITLE
72706 Create ivc_champva Module

### PR DESCRIPTION
**Pull request by balexandr**
## Summary
When the IVC team first came together we leveraged SimpleFormsAPI module to hit the ground running. Now that we have a better understanding of the code and begin to branch off of SFA more and more it made sense to create our own module.

- Utilized `vtk` to generate the scaffolding for the new `ivc_champva` module
- Ported the code over from SimpleFormsAPI and unravelled the logic
- Cleaned up a lot of the code and made it a bit more readable

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/72706

## Testing done

- Tested the new route `/ivc_champva/v1/forms` in the browser and saw the files uploaded in S3
- Tested the new route `/ivc_champva/v1/forms/submit_supporting_documents` in the browser as well
- Updated and ran Rspec tests

## What areas of the site does it impact?
- This is a brand new module so it shouldn't effect any other aspects of vets-api

### CODE REVIEW

**Review by michaelclement — APPROVED** [review]
Looks good, lines up with previous implementation progress. It will be good to have a separate IVC module so we're not stepping on toes.

**Review by cloudmagic80 — DISMISSED** [review]
looks okay. I ran it locally

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'ddtrace'
+
+module IvcChampva
+  module V1
+    class UploadsController < ApplicationController
+      skip_before_action :authenticate
+      before_action :authenticate, if: :should_authenticate
```
Comment (stevenjcumming) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:9]: If you intend `should_authenticate` to always be true, then you don't need to have `if: :should_authenticate`

Also it's not necessary to have `skip_before_action :authenticate`

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          attachment = PersistentAttachments::MilitaryRecords.new(form_id: params[:form_id])
+          attachment.file = params['file']
+          raise Common::Exceptions::ValidationErrors, attachment unless attachment.valid?
+
+          attachment.save
+          render json: attachment
+        end
+      end
+
+      def authenticate
```
Comment (stevenjcumming) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:43]: This should be a private method

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**